### ISSUE TITLE
Issue with file lock when creating image

### ISSUE THREAD

**Issue by tfenster** [microsoft/navcontainerhelper#1537]
**Describe the issue**
When creating an image with New-BCImage, we sometimes see this

```
stderr: Set-Content : The process cannot access the file
stderr: 'C:\bcartifacts.cache\sandbox\17.1.18256.19901\de\lastused' because it is
stderr: being used by another process.
stderr: At C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\1.0.16\Misc\Dow
stderr: nload-Artifacts.ps1:112 char:13
stderr: + Set-Content -Path (Join-Path $appArtifactPath 'lastused') ...
stderr: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
stderr: + CategoryInfo : NotSpecified: (:) [Set-Content], IOException
```

Is that a known issue? Would a try / catch / retry make sense here?

**Scripts used to create container and cause the issue**
```
New-BcImage -artifactUrl (Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox -version 17) -imageName bcartifacts/cosmo-bc -skipIfImageAlreadyExists -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit -licenseFile <removed>
```

**Full output of scripts**
```
stdout: BcContainerHelper version 1.0.16
stderr: Set-Content : The process cannot access the file 
stderr: 'C:\bcartifacts.cache\sandbox\17.1.18256.19901\de\lastused' because it is 
stderr: being used by another process.
stderr: At C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\1.0.16\Misc\Dow
stderr: nload-Artifacts.ps1:112 char:13
stderr: +             Set-Content -Path (Join-Path $appArtifactPath 'lastused') ...
stderr: +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
stderr:     + CategoryInfo          : NotSpecified: (:) [Set-Content], IOException
stderr:     + FullyQualifiedErrorId : System.IO.IOException,Microsoft.PowerShell.Comma 
stderr:    nds.SetContentCommand
```

**Additional context**
- does it happen all the time?
No
- did it use to work?
Yes

It happens on one of two machines which are automatically provisioned by a script and should be as close to identical as possible. On one it works, on the other it doesn't, so it seems to be some kind of race condition or corner case to me

**Comment by freddydk** [comment]
I will make that part more resilient to allow for this scenario to ignore the 2nd part.

**Comment by freddydk** [comment]
This was strange.
There is a mutex around these - they should not be able to ever run at the same time, but apparently, Set-Content is slow at releasing the lock on files, which is what caused the issue.

**Comment by tfenster** [comment]
Thanks!

**Comment by freddydk** [comment]
BTW - The code for downloading artifacts also exists inside the container. I will deploy a new generic image with the same fix.

**Comment by freddydk** [comment]
Shipped in 1.0.17
